pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java

```diff
@@ -95,14 +95,25 @@ public void update(ResourceGroup resourceGroup) {
                 this.publishMaxMessageRate = Math.max(resourceGroup.getPublishRateInMsgs(), 0);
                 this.publishMaxByteRate = Math.max(resourceGroup.getPublishRateInBytes(), 0);
                 if (this.publishMaxMessageRate > 0) {
-                    // TODO: pass the executor
-                    publishRateLimiterOnMessage =
-                        new RateLimiter(publishMaxMessageRate, 1, TimeUnit.SECONDS, this::apply);
+                    publishRateLimiterOnMessage = RateLimiter.builder()
+                            .scheduledExecutorService(scheduledExecutorService)
+                            .permits(publishMaxMessageRate)
+                            .rateTime(1L)
+                            .timeUnit(TimeUnit.SECONDS)
+                            .isDispatchOrPrecisePublishRateLimiter(false)
```

Comment:
the default value is false, so there's no need to set it

Reply:
done